src/DataStructures/DataBox/DataBoxTag.hpp

```diff
@@ -254,7 +254,7 @@ template <typename Tag, typename... Tags, typename... TagTypes>
 struct check_compute_item_is_invokable<Tag, tmpl::list<Tags...>,
                                        tmpl::list<TagTypes...>> {
   static_assert(
-      is_function_callable_v<Tag, TagTypes...>,
+      is_get_callable_v<Tag, TagTypes...>,
```

Comment:
I still don't quite understand the implications of this PR, and I think most of my confusion comes from symbol names that may just not be updated yet:

- We have "simple/compute/reference/base tags" now, where compute/reference tags are "immutable". Are simple and base tags treated separately from the "mutable/immutable" concept?
- Code that refers to either compute or reference tags is still named "compute" as far as I can tell, which is probably my main source of confusion. Should the struct I'm commenting on here and the one below carry "reference" in their name instead, and also say "reference item" in the error message?
- If I understand correctly, the change here and the one in DataBox.hpp:459(new) mean that compute items _must_ return by ref now. That seems the important thing to add to the upgrade instructions. (The renamed type aliases aren't used much anyway, so not sure if you even need to add them to the upgrade instructions.)
- The code around DataBox.hpp:459 has "mutable", "immutable" and "compute" in names, though if my understanding is correct it should only distinguish "compute" and "reference" (both of which are "immutable" by the definition of tags, though of course I understand that compute tags actually mutate storage in the DataBox so the choice of name might be confusing or should at least be commented on where it's relevant).

Reply:
There are two longer PRs coming next that should address all of these.  The first rewrites the implementation and the second updates the overall documentation.
-  Mutable vs immutable distinguish those items that can or cannot be mutated by calling `db::mutate`.  Compute and reference items are immutable; all others (which at the moment are simple items) are mutable.  
- All compute (reference) tags must derive from `db::ComputeTag` (`db::ReferenceTag`) and a simple tag.
- All simple tags must derive from `db::SimpleTag`.  A simple tag may also (in rare cases where type-erasure is needed) derive from a base tag (which itself must derive from `db::BaseTag`).
- The PR rewriting the implementation will also clean up the internal names, addressing your 2nd and 4th bullets.  
- Yes, your third bullet is true, I've modified the upgrade instructions

Reply:
Ok I'll wait for those next PRs then :)